scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/AvroBucketMetadata.java

```diff
@@ -178,25 +178,45 @@ public void populateDisplayData(Builder builder) {
 
   @Override
   public boolean isPartitionCompatibleForPrimaryKey(BucketMetadata o) {
-    if (o == null || getClass() != o.getClass()) return false;
-    AvroBucketMetadata<?, ?, ?> that = (AvroBucketMetadata<?, ?, ?>) o;
-    return getKeyClass() == that.getKeyClass() && keyField.equals(that.keyField);
+    if (o == null) return false;
```

Comment:
todo - make this generic somehow -- i.e. some kind of method `public boolean Set<Class<? extends BucketMetadata>> isCompatibleWith()` ?